### ISSUE TITLE
`ExecTx` and `Exec` return type mismatch

### ISSUE THREAD

**Issue by vilhelmmelkstam** [encoredev/encore#55]
The following two functions gives no linting errors, but it cannot compile using encore.
```go
func DummyDbQuery1(ctx context.Context) (int64, error) {
	res, err := sqldb.Exec(ctx, `INSERT INTO foo (bar) VALUES (1)`)
	if err != nil {
		return 0, err
	}

	rowsAffected, err := res.RowsAffected()
	return rowsAffected, err
}

func DummyDbQuery2(ctx context.Context) (int64, error) {
	res, err := sqldb.Exec(ctx, `INSERT INTO foo (bar) VALUES (1)`)
	if err != nil {
		return 0, err
	}

	lastInsertedId, err := res.LastInsertId()
	return lastInsertedId, err
}
```

The following errors are instead returned when I try to run or check the program:
```
customization/repository/customization.go:66:20: assignment mismatch: 2 variables but res.RowsAffected returns 1 values
customization/repository/customization.go:76:28: res.LastInsertId undefined (type sqldb.ExecResult has no field or method LastInsertId)
```

The error seems to be the difference between https://github.com/encoredev/encore/blob/786cf64412e55c8ac5a829198efa61132a800a97/compiler/runtime/storage/sqldb/sqldb.go#L199 and https://github.com/encoredev/encore.dev/blob/0069c0de93b0c50cc8b9c33ceb61e08000f07f54/storage/sqldb/sqldb.go#L65. (`ExecResult` vs `sql.Result`)

### Environment
```bash
$ go version
go version go1.16.6 darwin/amd64
```
```bash
$ encore version
encore version v0.16.2
```

**Comment by eandre** [comment]
Thanks for the report, you're right. We used to use `sql.Result` but the issue is that PostgreSQL doesn't actually report the last inserted id from `INSERT` statements, see [INSERT docs](https://www.postgresql.org/docs/current/sql-insert.html#id-1.9.3.152.7). As a result we moved away from returning `sql.Result` and opted to only provide the `RowsAffected` method. 

The way to do what you want is by using `QueryTx`/`QueryRowTx` and structuring your query as `INSERT ... RETURNING id` or whatever column you want to retrieve.

I'll fix the signature in `encoredev/encore.dev` to match this. Thank you!
